python/tank/util/login.py

```diff
@@ -65,35 +68,47 @@ def get_shotgun_user(sg):
         g_shotgun_user_cache = sg.find_one("HumanUser", filters=[["login", "is", local_login]], fields=fields)
     
     return g_shotgun_user_cache
-        
+
 def get_current_user(tk):
     """
-    Retrieves the current user as a dictionary of metadata values.
-    Returns None if the user is not found in shotgun.
-    
-    Returns the following fields:
-    
-    * id
-    * type
-    * email
-    * login
-    * name
-    * image url (thumbnail)
-    
-    This method connects to shotgun.
-    """    
+    Retrieves the current user as a dictionary of metadata values. Note: This method connects to
+    shotgun the first time around. The result is then cached to reduce latency.
+    :returns: None if the user is not found in shotgun. Otherwise, it returns a dictionary
+              with the following fields:
+                 * id
+                 * type
+                 * email
+                 * login
+                 * name
+                 * image url (thumbnail)
+    """
     global g_shotgun_current_user_cache
-    if g_shotgun_current_user_cache == "unknown":
-    
+    if g_shotgun_current_user_cache != "unknown":
+        return g_shotgun_current_user_cache
+
+    from .. import api
+
+    user = api.get_current_user()
```

Comment:
if we remove `get_current_user` from the public API, I guess this could access the global directly. This may also be a little more clear – i find it a tad confusing that this `get_current_user` method is calling another `get_current_user` method somewhere else...